utility/drivers.cpp

```diff
@@ -1,120 +1,68 @@
 /**
- * @file   drivers.c
+ * @file   drivers.cpp
  * @Author Ben Peters (ben@ardusat.com)
+ * @Author Sam Olds (sam@ardusat.com)
  * @date   December 3, 2014
  * @brief  Implements sensor-specific driver read/initialization functions
  *
  * These functions are meant to provide a gateway interface that the higher-level
  * generic interface uses to get data from individual sensors.
  */
 
-#include <Arduino.h>
-#include <Wire.h>
-#include "drivers.h"
-#include "pololu_LSM303.h"
 #include "ArdusatSDK.h"
+#include "drivers.h"
+#include <Wire.h>
 
 static config_lsm303_accel_t _lsm303_d_accel_config;
 static config_lsm303_mag_t _lsm303_d_mag_config;
 static config_l3gd20_t _l3gd20_config;
 
 /**
- * Generic I2C Read function
- *
- * @param devAddr I2C Device address
- * @param reg Register address
- * @param val pointer to byte array to read into
- * @param length number of bytes to read
- *
- * @return 0 on success, other on failure
+ * Check to see if both the ISL29125 and TCS34725 RGB Light Sensors
+ * exists. They are only included with the spaceboard, not the space
+ * kit, so if they do exist, the user has a spaceboard.
  */
-int _readFromRegAddr(uint8_t devAddr, uint8_t reg, void *val, int length)
-{
-  uint8_t *byteArray = (uint8_t *) val;
-  int ret;
-  int readData = 0;
-
-  Wire.beginTransmission(devAddr);
-
-  if (Wire.write(reg) < 1)
-    return -1;
-
-  if ((ret = Wire.endTransmission(false)) != 0) {
-    return ret;
-  }
-
-  if (byteArray == NULL || length == 0)
-    return 0;
-
-  Wire.beginTransmission(devAddr);
-  Wire.requestFrom((byte) devAddr, (byte) length);
-
-  while (Wire.available() > 0) {
-    byteArray[readData++] = Wire.read();
-  }
-
-  if ((ret = Wire.endTransmission(true)) != 0) {
-    return ret;
+void catchSpaceboard() {
+  // Only need to check once, as `catchSpaceboard` is called by all
+  // sensor `begin` functions
+  if (!ARDUSAT_SPACEBOARD && !MANUAL_CONFIG) {
+    uint8_t islData;
+    uint8_t tcsData;
+    Wire.begin();
+
+    // Here, we are checking to see if the sensor replies with the
+    // correct response at it's expected address on the spaceboard
+    readFromRegAddr(DRIVER_SPACEBOARD_ISL29125_ADDR, 0x00, &islData, 1, BIG_ENDIAN);
+    readFromRegAddr(DRIVER_SPACEBOARD_TCS34725_ADDR, 0x80 | 0x12, &tcsData, 1, BIG_ENDIAN);
+
+    // Checking the responses from the identify registers of each
+    // sensor to make sure it's what we expect for both
```

Comment:
I actually think this is correct. "the identify registers of each sensor to make sure _it is_ what we expect"